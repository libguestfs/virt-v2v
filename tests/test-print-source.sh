#!/bin/bash -
# libguestfs virt-v2v test script
# Copyright (C) 2014 Red Hat Inc.
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

# Test --print-source option.

set -e

source ./functions.sh
set -e
set -x

skip_if_skipped
requires test -f ../test-data/phony-guests/windows.img

export LANG=C

d=test-print-source.d
rm -rf $d
cleanup_fn rm -r $d
mkdir $d

$VG virt-v2v --debug-gc \
    -i libvirtxml test-print-source.xml \
    -o local -os $d \
    --print-source > $d/output

mv $d/output $d/output.orig
< $d/output.orig \
grep -v 'Setting up the source' |
grep -v 'Opening the source' |
grep -v 'Source guest information' |
sed -e 's,/.*/windows.img,windows.img,' |
grep -v '^$' \
> $d/output

diff -u "$srcdir/test-print-source.expected" $d/output
