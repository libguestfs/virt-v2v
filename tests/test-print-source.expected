    source name: windows
hypervisor type: kvm
       VM genid: 
         memory: 1073741824 (bytes)
       nr vCPUs: 64
     CPU vendor: Intel
      CPU model: Broadwell
   CPU topology: sockets: 4 cores/socket: 8 threads/core: 2
   CPU features: acpi,apic,pae
       firmware: unknown
        display: 
          sound: 
disks:
	0 [virtio-blk]
removable media:
NICs:
	Network "default" mac: 00:11:22:33:44:55 [virtio]
